extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/HttpServerOptionsUtils.java

```diff
@@ -428,4 +431,23 @@ private static void setIdleTimeout(HttpConfiguration httpConfiguration, HttpServ
         options.setIdleTimeout(idleTimeout);
         options.setIdleTimeoutUnit(TimeUnit.MILLISECONDS);
     }
+
+    public static HttpConfiguration.InsecureRequests getInsecureRequestStrategy(HttpBuildTimeConfig buildTimeConfig,
+            Optional<HttpConfiguration.InsecureRequests> requests) {
+        if (requests.isPresent()) {
+            var value = requests.get();
+            if (buildTimeConfig.tlsClientAuth == ClientAuth.REQUIRED && value == HttpConfiguration.InsecureRequests.ENABLED) {
+                throw new ConfigurationException(
```

Comment:
@cescoffier 

Should we start with `warn` in cases when the users decided to enable it ? I'm thinking if they decided to do it then there might be a good reason for it, one case I can imagine, which may be rare enough, but one can restrict MTLS Required to specific paths, 

https://quarkus.io/guides/security-authentication-mechanisms#path-specific-authentication-mechanisms

for example:
```
quarkus.http.auth.permission.mtls.paths=/mtls-required
quarkus.http.auth.permission.mtls.policy=authenticated
quarkus.http.auth.permission.mtls.auth-mechanism=mtls
```

and do something else for other paths. I have to admit, I can't really imagine why, in prod, one would have a semi-secured application like that, where the client certificate is enforced on some paths, and then anyone can access that app over plain HTTP on some other paths.

But may be `warn` would be a bit softer, and then if users won't push back a lot on this warning in 3.7.0.CR1/3.7.0, we can totally enforce as you propose with this exception in 3.8.0, what do you think ?

Reply:
It should fix the failing tests